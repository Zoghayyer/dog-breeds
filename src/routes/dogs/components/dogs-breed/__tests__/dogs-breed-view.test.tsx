import React from 'react';
import { shallow } from 'enzyme';
import DogsBreedView, { DogsBreedViewProps } from '../dogs-breed-view';

describe('(View) DogsBreedView', () => {
  let component;
  const defaultProps: DogsBreedViewProps = {
    dogs: [
      {
        alt: 'dog number 0',
        favorite: false,
        id: 0,
        url: 'fake-url-0',
      },
      {
        alt: 'dog number 1',
        favorite: true,
        id: 1,
        url: 'fake-url-1',
      },
    ],
    handleClickFavorite: () => ({}),
  };

  const shallowDogsBreedView = (extendedProps: Partial<DogsBreedViewProps> = {}) => {
    component = shallow(<DogsBreedView {...defaultProps} {...extendedProps} />);
  };

  beforeEach(() => shallowDogsBreedView());

  it('renders DogBreed component for every dog in the dogs array', () => {
    expect(component).toMatchSnapshot();
  });
});
