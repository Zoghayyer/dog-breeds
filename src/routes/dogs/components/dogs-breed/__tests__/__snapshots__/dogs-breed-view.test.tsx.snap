// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(View) DogsBreedView renders DogBreed component for every dog in the dogs array 1`] = `
<div
  className="mt-4 image-card"
>
  <DogBreed
    dog={
      Object {
        "alt": "dog number 0",
        "favorite": false,
        "id": 0,
        "url": "fake-url-0",
      }
    }
    handleClickFavorite={[Function]}
    key="0-fake-url-0"
  />
  <DogBreed
    dog={
      Object {
        "alt": "dog number 1",
        "favorite": true,
        "id": 1,
        "url": "fake-url-1",
      }
    }
    handleClickFavorite={[Function]}
    key="1-fake-url-1"
  />
</div>
`;
